dss/events/handlers/sync.py

```diff
@@ -1,20 +1,35 @@
+import os
+import sys
 import datetime
+import json
 import time
 import hashlib
 from contextlib import closing
+from concurrent.futures import ThreadPoolExecutor
+from collections import namedtuple
 
 import boto3
 import botocore.session
 import urllib3
 import google.cloud.storage
+from google.resumable_media._upload import get_content_range
 from google.cloud.client import ClientWithProject
 from google.cloud._http import JSONConnection
 
 from dss import Config
+from dss.util.aws import resources, clients, send_sns_msg, ARN
+from dss.util.streaming import get_pool_manager, S3SigningChunker
+from dss.blobstore.gs import GSBlobStore
 
 presigned_url_lifetime_seconds = 3600
 use_gsts = False
 gsts_sched_delay_minutes = 2
+part_size = {"s3": 64 * 1024 * 1024, "gs": 64 * 1024 * 1024 * 10}
```

Comment:
The rationale for the gs part size here was to basically apportion an amount of work that one Lambda can reasonably be expected to complete.

As it stands, there is nothing that guarantees that we stay under the GS 1024 component count limit. There is an implicit limit of 640 GB (ought to be enough for everyone 😁), beyond which GS uploads will fail to close.

I agree with you regarding the strategy to determine part size. It seems clear that a single lambda won't be able to copy (5TB/1024) = 5GB of data before running out of the allotted 5 minutes. If AWS doesn't increase the limit soon, and there is no hidden inefficiency that I overlooked, then I'll want to do Lambda chaining (a Lambda senses that it's about to run out of time and relaunches a copy of itself - a pattern I've successfully used before) on each part's resumable upload.

Filed #275 to track part size optimization and tracking against the part count limits.

Fixed this to read `640 * 1024 * 1024`.